tools/purge_omsagent.sh

```diff
@@ -0,0 +1,407 @@
+echo "--------------------------------------------------------------------------------"
+echo ""
+
+# Check if installed via extension
+ls /var/lib/waagent | grep -i Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux- > /dev/null 2>&1
+EXTENSION_INSTALLED=$?
+if [ $EXTENSION_INSTALLED -eq 0 ]; then
+    echo ""
+    echo "Please go to portal.azure.com and uninstall the OMS extension and its dependencies."
+    echo "Go to Azure Portal -> Virtual Machines -> <vm_name> -> Settings -> Extensions"
+    echo "and then click the '...' in the OMSAgentForLinux row and click 'Uninstall'"
+    echo "(Do this as well with the DependencyAgentLinux row)"
+    read -p "Press enter to proceed once the Extension is uninstalled." toss2
+    echo ""
+fi
+
+# run OMS purge command
+echo "---------- Downloading OMS Onboarding Script ----------"
+if [ ! -f $OMS_ONBOARD_AGENT ]; then
+    wget https://raw.githubusercontent.com/Microsoft/OMS-Agent-for-Linux/master/installer/scripts/onboard_agent.sh
+    if [ $? -ne 0 ]; then
+        echo "Error accessing Github to download the purge script for the OMS Agent" >&2
+        echo "Please ensure that this machine can connect to the following URL:" >&2
+        echo "  https://raw.githubusercontent.com/Microsoft/OMS-Agent-for-Linux/master/installer/scripts/onboard_agent.sh" >&2
+        echo "Or ensure the onboard_agent.sh script is available in the same directory as this script." >&2
+        call_exit $ONBOARD_AGENT_MISSING
+    fi
+fi
+echo ""
+
+echo "---------- Running OMS Purge Script ----------"
+chmod +x ./onboard_agent.sh
+./onboard_agent.sh --purge
+
+# remove OMS Extension info
+
+if [ -f $WAAGENT_XML ]; then
+    STOPPED_WAAGENT=1
+    echo "Temporarily stopping waagent..."
+    stop_waagent
+
+    echo "Removing OMS Extension files..."
+    rm -rf $WAAGENT_XML
+fi
+
+# remove directories (if they exist)
+echo "---------- Removing all relevant directories ----------"
+dir_rm /etc/opt/microsoft/auoms auoms
+dir_rm /etc/opt/microsoft/dependency-agent dependency-agent
+dir_rm /etc/opt/microsoft/omsagent omsagent
+dir_rm /etc/opt/microsoft/scx scx
+if [ ! "$(ls -A /etc/opt/microsoft > /dev/null 2>&1)" ]; then
+    dir_rm /etc/opt/microsoft
+fi
+
+dir_rm /opt/microsoft/auoms auoms
+dir_rm /opt/microsoft/dependency-agent dependency-agent
+dir_rm /opt/microsoft/omsagent omsagent
+dir_rm /opt/microsoft/omsconfig omsconfig
+dir_rm /opt/microsoft/scx scx
+if [ ! "$(ls -A /opt/microsoft > /dev/null 2>&1)" ]; then
+    dir_rm /opt/microsoft
+fi
+
+dir_rm /var/opt/microsoft/auoms auoms
+dir_rm /var/opt/microsoft/dependency-agent dependency-agent
+dir_rm /var/opt/microsoft/omsagent omsagent
+dir_rm /var/opt/microsoft/omsconfig omsconfig
+dir_rm /var/opt/microsoft/scx scx
+if [ ! "$(ls -A /var/opt/microsoft > /dev/null 2>&1)" ]; then
+    dir_rm /var/opt/microsoft
+fi
+
+if [ $EXTENSION_INSTALLED -eq 0 ]; then
+    dir_rm /var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-*
```

Comment:
1. doesn't `dir_rm` take 2 parameters? 
2. not an expert on wildcard expansion but pretty sure this will expand to all possible extension versions _before_ dir_rm is executed, which I would expect to then give an error from dir_rm since param 2 would not be a package name. we want this to remove all possible versions, though, so either dir_rm needs to be changed to take package name first and then >= 1 dirpath to remove, or we need a loop here to iterate over the wildcard expansion

Reply:
1. It's a weird workaround but since the second parameter is empty it gets passed to `pkg_rm`, which has a check to make sure that the input isn't empty (or if it is empty, it does nothing)
2. I think a loop would probably work best, I'll make that change

Reply:
Both fixed in this commit: https://github.com/microsoft/OMS-Agent-for-Linux/pull/1255/commits/1b4155724692379c7c06374f903f0109755cd418